### DISCUSSION TITLE
Where is difference between `#[bw]` `#[br]` and `#[brw]` articulated in the docs?

### DISCUSSION THREAD

**Discussion by vi** [jam1garner/binrw#200] [Q&A]
I see `#[binread]`, `#[binrw]` and `#[binwrite]` attribute macros all link to `binrw::docs::attribute` table, which contains `r/w` column.
Also I see that `derive(BinRead)` things use `#[br]` and `derive(BinWrite)` things use `#[bw]` in examples.

Why they are distinct attributes? Shall this connection from distinct attribute or derive macros to derive helpers and to "r/w" column in the table be documented explicitly?

**Comment by csnover** [comment]
Attributes are distinct because reading and writing are distinct. Taking a `u32` to/from `usize`, you need to `usize::try_from` for reading and `u32::try_from` for writing. Taking a `Vec<T>`, you need to take the vector length from the stream when reading and you need to take the length from `v.len()` when writing. Also, read values are owned and write values are borrowed.

Was there a specific thing you were trying to do that caused this confusion?
